app/Http/Controllers/CartsController.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Product;
+use App\Models\Cart;
+use App\Moddel\Category;
+use Session;
+
+class CartsController extends Controller
+{    
+    public function addToCart(Request $request) 
+    {
+        try
+        {
+            $id = $request->id;
+            $product = Product::findOrFail($id);
+            $quantity = $request->quantity;
+            $image = $product->images[config('setting.zero_value')];
+            
+            $oldCart = Session::has('cart')?Session::get('cart'):null;
+            $cart = new Cart($oldCart);
+            $cart->add($product, $product->id, $quantity);
+
+            $request->session()->put('cart', $cart);
+            
+            return response()->json([
+                'cart' => $cart->items,
+                'totalQuantity' => $cart->totalQuantity,
+                'totalPrice' => $cart->totalPrice,
+                'product' => $product,
+                'image' => $product->images[config('setting.zero_value')]
+            ]);
+        }
+        catch (\Exception $e)
+        {
+            
```

Comment:
Không có thông báo gì hết à? :(

Reply:
Dạ em bắt exception rồi dùng ajax để báo lỗi ra thôi ạ.
error:function(){
            $('#error_msg').html('Product not found!');
        }